### PR TITLE
refactored github issue templates

### PR DESCRIPTION
**Pull request by shibumi**
*Please fill in the fields below to submit a pull request.  The more
information that is provided, the better.*

**Fixes issue #**: None

**Description of pull request**:

This PR changes the issue templating. I tried to orientate myself on sigstore's issue style:

**Please verify and check that the pull request fulfills the following
requirements**:

- [X] Tests have been added for the bug fix or new feature
- [X] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by shibumi** [comment]
With this new approach we can also set labels and assignes. @adityasaky  @SantiagoTorres  do we want any other issue templates? Questions? Organization? etc?

Full documentation for this can be found here: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

File: .github/ISSUE_TEMPLATE/mentorship.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+labels: mentorship
+assignees: ''
+
+---
+
+<!--
+If you're new to the project, welcome!
+
+In addition to filling out this mentorship request, consider joining the
+in-toto slack community and introducing yourself!
```
Comment (adityasaky) [inline .github/ISSUE_TEMPLATE/mentorship.md:15]: Can we point to the CNCF slack workspace here?

**Review by adityasaky — CHANGES_REQUESTED** [review]
Looks great! I think we should also consider this on the Python implementation at the very least. One change though. :)

File: .github/ISSUE_TEMPLATE/mentorship.md (hunk 1 above)
Comment (shibumi) [reply .github/ISSUE_TEMPLATE/mentorship.md:15]: It is possible to set these templates for the whole organization. The sigstore community is doing this. But if we do somebody else have to set it up.

Comment (shibumi) [reply .github/ISSUE_TEMPLATE/mentorship.md:15]: Yes, we can include a link there. Let me have a look on it tomorrow. It's late already in germany (2:38AM) :)

**Comment by shibumi** [comment]
@adityasaky  done.. I think the channel link should work..

File: .github/ISSUE_TEMPLATE/mentorship.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+https://app.slack.com/client/T08PSQ7BQ/CM46K2VT2
```
Comment (adityasaky) [inline .github/ISSUE_TEMPLATE/mentorship.md:17]: Trying to open this in a private tab asks me to sign in, with even the workspace URL not filled in.

File: .github/ISSUE_TEMPLATE/mentorship.md (hunk 1 above)
Comment (adityasaky) [reply .github/ISSUE_TEMPLATE/mentorship.md:15]:
```suggestion
#in-toto channel on the CNCF slack workspace and introducing yourself!
Link to sign up: https://slack.cncf.io/
```

**Review by adityasaky — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/mentorship.md (hunk 2 above)
Comment (adityasaky) [reply .github/ISSUE_TEMPLATE/mentorship.md:17]:
```suggestion

```

**Review by adityasaky — APPROVED** [review]

**Comment by adityasaky** [comment]
Thanks @shibumi! I took the liberty of updating the Slack link.

**Pull request merged**